src/test/java/org/candlepin/model/AnonymousCloudConsumerCuratorTest.java

```diff
@@ -0,0 +1,171 @@
+        assertThat(actual)
+            .singleElement()
+            .returns(expected.getId(), AnonymousCloudConsumer::getId)
+            .returns(expected.getUuid(), AnonymousCloudConsumer::getUuid)
+            .returns(expected.getCloudAccountId(), AnonymousCloudConsumer::getCloudAccountId)
+            .returns(expected.getCloudInstanceId(), AnonymousCloudConsumer::getCloudInstanceId)
+            .returns(expected.getCloudProviderShortName(), AnonymousCloudConsumer::getCloudProviderShortName)
+            .returns(expected.getProductId(), AnonymousCloudConsumer::getProductId);
+    }
+
+    @ParameterizedTest(name = "{displayName} {index}: {0}")
+    @NullSource
+    @ValueSource(strings = { "", "  " })
+    public void testGetByUuidWithInvalidUuid(String uuid) {
+        AnonymousCloudConsumer expected = new AnonymousCloudConsumer()
+            .setCloudAccountId("cloudAccountId")
+            .setCloudInstanceId("instanceId")
+            .setProductId("productId")
+            .setCloudProviderShortName("shortName");
+
+        this.anonymousCloudConsumerCurator.create(expected);
+
+        AnonymousCloudConsumer actual = this.anonymousCloudConsumerCurator.getByUuid(uuid);
+
+        assertNull(actual);
+    }
+
+    @Test
+    public void testGetByUuidWithNonExistingUuid() {
+        AnonymousCloudConsumer expected = new AnonymousCloudConsumer()
+            .setCloudAccountId("cloudAccountId")
+            .setCloudInstanceId("instanceId")
+            .setProductId("productId")
+            .setCloudProviderShortName("shortName");
+        this.anonymousCloudConsumerCurator.create(expected);
+
+        AnonymousCloudConsumer actual = this.anonymousCloudConsumerCurator.getByUuid(Util.generateUUID());
+
+        assertNull(actual);
+    }
+
+    @Test
+    public void testGetByUuidWithExistingUuid() {
+        AnonymousCloudConsumer expected = new AnonymousCloudConsumer()
+            .setCloudAccountId("cloudAccountId")
+            .setCloudInstanceId("instanceId")
+            .setProductId("productId")
+            .setCloudProviderShortName("shortName");
+        this.anonymousCloudConsumerCurator.create(expected);
+
+        AnonymousCloudConsumer actual = this.anonymousCloudConsumerCurator.getByUuid(expected.getUuid());
+
+        assertThat(actual)
+            .isNotNull()
+            .returns(expected.getId(), AnonymousCloudConsumer::getId)
+            .returns(expected.getUuid(), AnonymousCloudConsumer::getUuid)
+            .returns(expected.getCloudAccountId(), AnonymousCloudConsumer::getCloudAccountId)
+            .returns(expected.getCloudInstanceId(), AnonymousCloudConsumer::getCloudInstanceId)
+            .returns(expected.getCloudProviderShortName(), AnonymousCloudConsumer::getCloudProviderShortName)
+            .returns(expected.getProductId(), AnonymousCloudConsumer::getProductId);
+    }
+
+    @ParameterizedTest(name = "{displayName} {index}: {0}")
+    @NullSource
+    @ValueSource(strings = { "", "  " })
```

Comment:
This won't eliminate the need for the additional ValueSource here, but jUnit has a composite NullAndEmptySource that will cover the need to use a value source for the first value here.

Reply:
Updated throughout